markers/stream_excluder.py

```diff
@@ -71,3 +91,25 @@ def pytest_collection_modifyitems(session, config, items):
     # Just to print out the appliance's streams
     if not ssh_connection_failed:
         reporter(config).write("\nAppliance's streams: [{}]\n".format(", ".join(get_streams_id())))
+        # Bail out if the appliance stream or version do not match
+        check_stream = config.getvalue("check_stream")
```

Comment:
I think you should default check_stream to `''`, declare its type as `str`, and then `check_stream = config.getvalue("check_stream").lower()` here.